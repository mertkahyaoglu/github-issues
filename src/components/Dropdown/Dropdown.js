import React, { Component } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

export default class Dropdown extends Component {
  state = {
    isOpen: this.props.isOpen,
  };

  static propTypes = {
    isOpen: PropTypes.bool,
    title: PropTypes.string,
    data: PropTypes.array,
    renderListItem: PropTypes.func,
  };

  componentWillReceiveProps(nextProps) {
    const { isOpen } = this.props;
    if (nextProps.isOpen !== isOpen) {
      this.setState({ isOpen: nextProps.isOpen });
    }
  }

  closeDropdown = () => {
    this.setState({ isOpen: false });
  };

  render() {
    const { isOpen, title, data, renderListItem, onItemClick } = this.props;
    return (
      <DropdownHolder isOpen={isOpen}>
        <DropdownWrapper>
          <Header>
            {title} <CloseButton onClick={this.closeDropdown}>x</CloseButton>
          </Header>
          <List>
            {data.map(item => (
              <Item key={item.id} onClick={() => onItemClick(item)}>
                {renderListItem(item)}
              </Item>
            ))}
          </List>
        </DropdownWrapper>
      </DropdownHolder>
    );
  }
}

const DropdownHolder = styled.div`
  position: absolute;
  right: 10px;
  z-index: 100;
  display: ${props => (props.isOpen ? 'block' : 'none')};
`;

const DropdownWrapper = styled.div`
  position: relative;
  width: 300px;
  margin-top: 4px;
  margin-bottom: 20px;
  overflow: hidden;
  font-size: 12px;
  color: #586069;
  background-color: #fff;
  background-clip: padding-box;
  border: 1px solid rgba(27, 31, 35, 0.15);
  border-radius: 3px;
  box-shadow: 0 3px 12px rgba(27, 31, 35, 0.15);
`;

const Header = styled.div`
  padding: 8px 10px;
  line-height: 16px;
  background: #f6f8fa;
  border-bottom: 1px solid #e1e4e8;
  text-align: left;
  font-weight: 600;
`;

const CloseButton = styled.span`
  display: block;
  float: right;
  color: #c6cbd1;
  cursor: pointer;
  font-size: 15px;
`;

const List = styled.div`
  position: relative;
  max-height: 400px;
  overflow: auto;
`;

const Item = styled.a`
  text-align: left;
  background-color: #fff;
  display: block;
  padding: 8px 8px 8px 30px;
  overflow: hidden;
  color: inherit;
  cursor: pointer;
  border-bottom: 1px solid #eaecef;
`;
