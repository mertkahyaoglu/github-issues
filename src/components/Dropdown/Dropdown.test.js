import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Dropdown from './Dropdown';
import TestProviders from 'utils/TestProviders';

describe('component/Dropdown', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <TestProviders>
        <Dropdown
          isOpen={true}
          data={labels}
          title="Sort by label"
          onItemClick={f => f}
          renderListItem={item => {
            return <span>{item.name}</span>;
          }}
        />
      </TestProviders>,
      div
    );
  });

  it('renders markup correctly', () => {
    const tree = renderer
      .create(
        <TestProviders>
          <Dropdown
            isOpen={true}
            data={labels}
            title="Sort by label"
            onItemClick={f => f}
            renderListItem={item => {
              return <span>{item.name}</span>;
            }}
          />
        </TestProviders>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

const labels = [
  {
    id: 418438082,
    url: 'https://api.github.com/repos/jekyll/jekyll-admin/labels/stale',
    name: 'stale',
    color: 'bfd4f2',
    default: false,
  },
  {
    id: 418468082,
    url: 'https://api.github.com/repos/jekyll/jekyll-admin/labels/api',
    name: 'api',
    color: 'f6f6f6',
    default: false,
  },
];
