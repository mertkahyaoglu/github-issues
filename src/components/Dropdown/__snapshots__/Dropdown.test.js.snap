// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/Dropdown renders markup correctly 1`] = `
<div
  className="sc-bdVaJa eczwIr"
>
  <div
    className="sc-bwzfXH ZQJAD"
  >
    <div
      className="sc-htpNat hRQaLl"
    >
      Sort by label
       
      <span
        className="sc-bxivhb gIEwYI"
        onClick={[Function]}
      >
        x
      </span>
    </div>
    <div
      className="sc-ifAKCX bAgXMY"
    >
      <a
        className="sc-EHOje frrveb"
        onClick={[Function]}
      >
        <span>
          stale
        </span>
      </a>
      <a
        className="sc-EHOje frrveb"
        onClick={[Function]}
      >
        <span>
          api
        </span>
      </a>
    </div>
  </div>
</div>
`;
